adapters/repos/db/index_queue.go

```diff
@@ -539,15 +651,19 @@ func (q *IndexQueue) checkCompressionSettings() {
 		if err != nil {
 			q.Logger.WithError(err).Error("failed to turn on compression")
 		}
+
+		return true
 	}
+
+	return false
 }
 
 // pause indexing and wait for the workers to finish their current tasks
 // related to this queue.
 func (q *IndexQueue) pauseIndexing() {
 	q.Logger.Debug("pausing indexing, waiting for the current tasks to finish")
 	q.paused.Store(true)
-	q.queue.wait(q.ctx)
+	q.wait(context.Background())
```

Comment:
just wondering, why did we change `q.ctx` to `context.Background()`? and shouldn't we use here a `context.WIthTimeout` or is it safe to pass `context.Background()`?